### DISCUSSION TITLE
Avoid mutating a prop directly since the value will be overwritten whenever the parent component re-renders. Instead, use a data or computed property based on the prop's value. Prop being mutated

### DISCUSSION THREAD

**Discussion by khedri-zineddine** [nuxt/nuxt#19351] [Questions (Nuxt 2)] [answered]
i am getting this warning in all components where i pass props from parent to child , in child component i just read the props and display it in template and the props are always undefined

here is example how i am using props
import { Component, Vue, Prop } from 'vue-property-decorator';
```
@Component({
  name: 'PackItem',
  components: {},
  computed: {},
})
export default class PackItem extends Vue {
  @Prop() pack: any;
  @Prop() collapseDetails: boolean;
} 
```
i am using nuxtjs 2 with vue 2 and vue-property-decorator , i don't know if the issue is with the way i am sending props or what the issue exactly (because this component works properly in vue app but in nuxt app i it doesn't work it read all props as null)

**Comment by dsvgl** [comment, marked as answer]
Decorators start with capital letters.

So it is:
`@Component` and `@Prop`
instead of
`@component` and `@prop`

Typescript should give you red squiggly lines...

**Reply by khedri-zineddine** [reply]
i am already using  @Component and @Prop  when i post it comes lowercase , the issue isn't syntax , because same code works fine in vue but when moving these component to nuxt i start getting this issue

**Reply by dsvgl** [reply]
I also use nuxt2 with vue-property-decorator and vue-class-component. No issues.

Here are the versions I use:
```
"vue-class-component": "7.2.6",
"vue-property-decorator": "9.1.2",
```

To make sure, you really don't mutate props?

Maybe you could create two dummy hello-world components and see if you can recreate the error there too.

**Reply by khedri-zineddine** [reply]
resolved the issue , so the issue was in typescript version the default version for nuxt2 is 4.2.4 but when i was migrating my existing vue app it was work in some future of typescript 4.8.2 so vue-class-component with nuxt was complaining about this version, i downgrade the typescript make the nuxt app works fine.

**Reply by dsvgl** [reply]
ah, nice. Glad you solved it.
I once tried to upgrade Typescript from said 4.2.4, but everything broke. So I'm still on 4.2.4 ^^
